File: app/src/main/java/com/sabihamumcu/kelimekutusu/Send.java
package com.sabihamumcu.kelimekutusu;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.net.Uri;
import android.os.Environment;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Toast;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;
import com.sabihamumcu.kelimekutusu.model.Word;
import com.sabihamumcu.kelimekutusu.viewmodel.WordViewModel;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Send extends AppCompatActivity {

    ArrayList<String> gonderilecekListe = new ArrayList<>();
    Button b;
    private WordViewModel wordViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_send);
        wordViewModel = ViewModelProviders.of(this).get(WordViewModel.class);

        b = (Button) findViewById(R.id.anaEkranaDon);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Send.this, MainActivity.class);
                startActivity(intent);
            }
        });

        File path = Send.this.getDatabasePath("db_words.db");
        wordViewModel.getWords().observe(this, new Observer<List<Word>>() {
            @Override
            public void onChanged(@Nullable List<Word> words) {
                for (int i = 0; i < words.size(); i++) {
                    gonderilecekListe.add(words.get(i).getWord() + " " + words.get(i).getDescription());
                }
                if (gonderilecekListe.isEmpty()) {
                    Toast.makeText(Send.this, "You have an empty word box.", Toast.LENGTH_SHORT).show();
                }
            }
        });


        try {
            String fileName = "output.txt";
            File root = new File(Environment.getExternalStorageDirectory(), "testDir");
            if (!root.exists()) {
                root.mkdirs();
            }

            File gpxfile = new File(root, fileName);
            FileWriter writer = new FileWriter(gpxfile);
            for (int i = 0; i < gonderilecekListe.size(); i++) {
                writer.write(gonderilecekListe.get(i));
                writer.write("\n");
            }
            writer.flush();
            writer.close();
            sendEmail(gpxfile);
        } catch (IOException e) {
            e.printStackTrace();
        }

    }

    protected void sendEmail(File file) {
        Uri path = Uri.fromFile(file);

        Intent i = new Intent(Intent.ACTION_SEND);
        i.setType("message/rfc822");
        i.putExtra(Intent.EXTRA_SUBJECT, "Kelime Kutusu");
        i.putExtra(Intent.EXTRA_TEXT, "Kelime Kutunuz");
        i.putExtra(Intent.EXTRA_STREAM, path); // Include the path
        try {
            startActivity(Intent.createChooser(i, "Send mail..."));
        } catch (android.content.ActivityNotFoundException ex) {
            Toast.makeText(this, "There are no email clients installed.", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/sabihamumcu/kelimekutusu/Send.kt
package com.sabihamumcu.kelimekutusu

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.os.Environment
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.sabihamumcu.kelimekutusu.Send
import com.sabihamumcu.kelimekutusu.viewmodel.WordViewModel
import kotlinx.android.synthetic.main.activity_send.*
import java.io.File
import java.io.FileWriter
import java.io.IOException
import java.util.*

class Send : AppCompatActivity() {

    var gonderilecekListe = ArrayList<String>()

    private var wordViewModel: WordViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_send)
        wordViewModel = ViewModelProviders.of(this).get(WordViewModel::class.java)
        anaEkranaDon!!.setOnClickListener {
            val intent = Intent(this@Send, MainActivity::class.java)
            startActivity(intent)
        }
        val path = getDatabasePath("db_words.db")
        wordViewModel!!.words?.observe(this, Observer { words ->
            for (i in words!!.indices) {
                gonderilecekListe.add(words[i]?.word + " " + words[i]?.description)
            }
            if (gonderilecekListe.isEmpty()) {
                Toast.makeText(this@Send, "You have an empty word box.", Toast.LENGTH_SHORT).show()
            }
        })
        try {
            val fileName = "output.txt"
            val root = File(Environment.getExternalStorageDirectory(), "testDir")
            if (!root.exists()) {
                root.mkdirs()
            }
            val gpxfile = File(root, fileName)
            val writer = FileWriter(gpxfile)
            for (i in gonderilecekListe.indices) {
                writer.write(gonderilecekListe[i])
                writer.write("\n")
            }
            writer.flush()
            writer.close()
            sendEmail(gpxfile)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    protected fun sendEmail(file: File?) {
        val path = Uri.fromFile(file)
        val i = Intent(Intent.ACTION_SEND)
        i.type = "message/rfc822"
        i.putExtra(Intent.EXTRA_SUBJECT, "Kelime Kutusu")
        i.putExtra(Intent.EXTRA_TEXT, "Kelime Kutunuz")
        i.putExtra(Intent.EXTRA_STREAM, path) // Include the path
        try {
            startActivity(Intent.createChooser(i, "Send mail..."))
        } catch (ex: ActivityNotFoundException) {
            Toast.makeText(this, "There are no email clients installed.", Toast.LENGTH_SHORT).show()
        }
    }
}